.github/workflows/unit_tests.yml

```
@@ -54,6 +54,13 @@ jobs:
       - name: Install core repo
         run: |
           uv pip install --system -e .
-      - name: Run unittests
+      - name: Run unittests with coverage
         run: |
-          pytest --cov=json_database --cov-report xml test
+          pytest --cov=json_database --cov-fail-under=80 --cov-report=term-missing --cov-report=xml test
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
```

actionlint (1.7.11)

[error] the runner of "codecov/codecov-action@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)